php-frontend/src/main/java/org/sonar/plugins/php/api/visitors/CheckContext.java

```diff
@@ -36,25 +37,34 @@
 
   /**
    *
-   * <p> To set line and cost use {@link Issue#line(int)} and {@link Issue#cost(double)}. Note, that these calls could be chained.
+   * <p> To set line and cost use {@link PHPIssue#line(int)} and {@link PHPIssue#cost(double)}. Note, that these calls could be chained.
    * <pre>
-   *   newIssue("S000", "Some message")
+   *   newIssue(myCheck, "Some message")
    *     .line(105)
    *     .cost(3);
    * </pre>
    *
    * @param check the instance of the rule for which issue should be created
    * @param message message of the issue
    * @return instance of Issue
+   *
+   * @deprecated since 2.9. Use {@link CheckContext#newIssue(PHPCheck, Tree, String)}, {@link CheckContext#newLineIssue(PHPCheck, int, String)} or
+   * {@link CheckContext#newFileIssue(PHPCheck, String)}
    */
+  @Deprecated
   Issue newIssue(PHPCheck check, String message);
 
+  // fixme Add docs
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1134)